eox_core/cms/views.py

```diff
@@ -129,6 +142,85 @@ def course_team(request, *args, **kwargs):
     return JsonResponse(data)
 
 
+@enable_course_management_view
+@login_required
+@ensure_csrf_cookie
+@require_http_methods(["GET"])
+def get_courses_by_course_regex(request, *args, **kwargs):
+    """
+    **Use Cases**
+        Get a list of course keys that match with the provided query string.
+        E.g:
+            course-v1:FP+DAT248x+2018*
+            course-v1:FP+DAT*
+            course-v1:FP*
+            *DAT248x+2018*
+
+
+    **Example Requests**:
+        GET <studio-url>/eox-core/management/get_courses?search=<query-string>
+
+    **Request Values**
+
+        * search: Query string of the desired course matching.
+
+    **Response Values**
+
+        * message: Status message of the operation.
+        * status: Global operaion status.
+            Failed, Success.
+        * courses: List containing the course/s key/s that match with the provided query string.
+    """
+
+    input_course_regex = request.GET.get('search')
+    if not input_course_regex:
+        data = {
+            'message': 'No course regex provided.',
+            'status': 'Failed'
+        }
+        return JsonResponse(data, status=status.HTTP_400_BAD_REQUEST)
+
+    match_courses = []
+    course_regex = get_course_regex(input_course_regex)
+
+    courses_iter, in_process_course_actions = get_courses_accessible_to_user(request)
+    active_courses, archived_courses = get_process_courses_list(courses_iter, in_process_course_actions)
+
+    if not active_courses:
+        data = {
+            'message': 'No courses were found.',
```

Comment:
Do we have the piping in place to offer this as translated strings?

Reply:
No, since we are not defining a strings catalog and translations folder. @felipemontoya